src/Diagnostics.RuntimeHost/Services/MonitoringService/AuoHealMonitoringService.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Diagnostics.Logger;
+using Diagnostics.RuntimeHost.Utilities;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace Diagnostics.RuntimeHost.Services
+{
+    public interface IAutoHealMonitoringService
+    {
+    }
+
+    /// <summary>
+    /// This service is responsible for logging auto heal settings at periodic intervals.
+    /// </summary>
+    public class AuoHealMonitoringService : IAutoHealMonitoringService
+    {
+        private readonly bool isEnabled;
+        private readonly int iterationDelayInSeconds;
+        private readonly string appHostConfigFilePath;
+
+        /// <summary>
+        /// Creates a new instance of AutoHealMonitoringService
+        /// </summary>
+        /// <param name="configuration">Configuration object</param>
+        /// <param name="environment">Environment object</param>
+        public AuoHealMonitoringService(IConfiguration configuration, IHostingEnvironment environment)
+        {
+            isEnabled = configuration.GetValue("AutoHealMonitoringSettings:IsEnabled", true);
+            iterationDelayInSeconds = configuration.GetValue("AutoHealMonitoringSettings:IterationDelayInSeconds", 30 * 60);
+            appHostConfigFilePath = Environment.GetEnvironmentVariable("APP_POOL_CONFIG");
+
+            if (environment.IsProduction() && isEnabled)
+            {
+                LogAutoHealSettings();
```

Comment:
>LogAutoHealSettings [](start = 16, length = 19)

I don't know this works but reading this code looks like mixing sync code with Async. Also, does this mean that constructor is blocked because we have a While loop in LogAutoHealSettings? 

Maybe a better approach is to use a timer and initialize the timer in constructor and set its interval equal to IterationDelayInSeconds.

The recommended way as suggested by ASP.NET core is to use IHostedService - [https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services)

Reply:
this method  "LogAutoHealSettings" is fire and forget Task (not sync code). And since there is no "await" statement, the constructor is not blocked.  (Verified during testing).

I dont see a need to over complicate the logic here.

Reply:
okay. That sounds good!